### PR TITLE
Grand v2 Refactoring 

### PR DESCRIPTION
**Pull request by TimothyGu**
Migrated from #1791.

### CODE REVIEW

**Comment by ForbesLindesay** [comment]
I think what we'll actually want to do is branch v1 pretty soon and then merge this into master (along with anything currently in v2)

**Comment by ForbesLindesay** [comment]
P.S. I want to deal with the merge conflicts anyway as some of them are new test cases, and most of the others should be easy to resolve.

File: bin/jade.js [hunk 1]
```diff
@@ -58,6 +59,17 @@ program.on('--help', function(){
 
 program.parse(process.argv);
 
+// locals given, parse them
+
+var locals;
+if (program.locals) {
+  if (exists(program.locals)) {
+    locals = JSON.parse(fs.readFileSync(program.locals));
```
Comment (alejonext) [inline bin/jade.js]: Well, if you're going to get is better to use the `require` ?

Comment (ForbesLindesay) [inline bin/jade.js]: I wouldn't bother reviewing the changes in this file too much.  There are lots of changes in master that will need merging with a few changes in this, so what's here at the moment isn't particularly representative of what I expect the finished product to look like.

Comment (alejonext) [inline bin/jade.js]: OK! It is not to bother ... I do it in the best of the spirits! is that your work is hard!

Comment (ForbesLindesay) [inline bin/jade.js]: Didn't mean that to come across as a complaint.  It's always good to get feedback, just thought if you were reviewing the code I didn't want you to waste your time :)

**Comment by TimothyGu** [comment]
#1972

**Pull request closed without merging**